### ISSUE TITLE
Issues with Orleans 7

### ISSUE THREAD

**Issue by gparissis** [OrleansContrib/OrleansDashboard#379]
I have a few issues with the dashboard that I didn't have with the previous version of Orleans.
First, there is no counters package anymore, so I can't see Silo's CPU and memory utilization.
Second, sometimes when I click on a grain, I get the message that there is no info received or something similar.
Third, grains that don't receive requests still report a req/sec value even after 10 minutes.

So is there a problem with Orleans 7 that you are aware of?

**Comment by gparissis** [comment]
A specific error message not I haven't noticed besides the second one.
The third is not an error message but not a reasonable interpretation of the received data. maybe? I don't understand how a grain reports req/sec, even when I know that is not the case.
I am aware that Orleans 7 introduced many changes and that when I upgraded, things started breaking, so I don't know if the Dashboard suffered from this upgrade also. 
I am sure by now that the Nuget package reporting CPU and memory usage isn't available anymore, so my question was general in case there is something else you noticed and are working on. 
By the way, are you going to fix this regarding CPU and memory counters in the dashboard?

P.S.: Will take screenshots next time I see something strange.

**Comment by gparissis** [comment]
I took few screenshots right now
![image](https://user-images.githubusercontent.com/6654954/206024939-d7cee98e-bbd1-4451-8311-4016e54c4b5f.png)
The number of activations for the ChargebackAPI grain is not correct, it should be 47, but it shows only 1. I know this because I fetch all the ids from the database, and I activate each grain with the Id.

The second screenshot
![image](https://user-images.githubusercontent.com/6654954/206025335-8d8eb8a2-f305-4f9d-88e7-be68f175f991.png)

It shows that this grain still accepts requests, but this is not the case anymore. The throughput is 0.45 because of the example I executed, but when I run a more demanding job, this number can go up to 120 req/sec and stay there for a long time.

**Comment by richorama** [comment]
(for my own notes) I am seeing this error when getting grain details
```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.NullReferenceException: Object reference not set to an instance of an object.
         at OrleansDashboard.Implementation.Helpers.GrainStateHelper.<>c.<GetGrainTypes>b__0_1(Type w)
         at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
         at System.Text.Json.Serialization.Converters.IEnumerableDefaultConverter`2.OnWriteResume(Utf8JsonWriter writer, TCollection value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryWrite(Utf8JsonWriter writer, TCollection value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
         at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
         at System.Text.Json.JsonSerializer.WriteCore[TValue](Utf8JsonWriter writer, TValue& value, JsonTypeInfo`1 jsonTypeInfo)
         at System.Text.Json.JsonSerializer.Serialize[TValue](Utf8JsonWriter writer, TValue value, JsonSerializerOptions options)
         at OrleansDashboard.DashboardMiddleware.WriteJson[T](HttpContext context, T content)
         at OrleansDashboard.DashboardMiddleware.WriteJson[T](HttpContext context, T content)
         at OrleansDashboard.DashboardMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.InvokeCore(HttpContext context, PathString matchedPath, PathString remainingPath)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

**Comment by jsmithtx** [comment]
> Third, grains that don't receive requests still report a req/sec value even after 10 minutes.

Seeing the same thing as well.. on all my grains!

**Comment by jsmithtx** [comment]
> > Second, sometimes when I click on a grain, I get the message that there is no info received or something similar.
> 
> Seeing the same thing as well.. on all my grains!

Ok, this appears to be resolved by property hosting configuration -- I'm in SF and on both Silo and ASPNET service, `HostSelf = true` and it worked. Before when they were blank, I did not have HostSelf on the ASPNET SF service.

**Comment by richorama** [comment]
fixed by #385, released in 7.0.2
